packages/grafana-ui/src/components/Monaco/types.ts

```diff
@@ -0,0 +1,66 @@
+export type CodeEditorChangeHandler = (value: string) => void;
+
+export interface CodeEditorProps {
+  value: string;
+  language: string;
+  width?: number | string;
+  height?: number | string;
+
+  readOnly?: boolean;
+  showMiniMap?: boolean;
+
+  /**
+   * Callback after the editor has mounted that gives you raw access to monaco
+   *
+   * @experimental - real type is: monaco.editor.IStandaloneCodeEditor
+   */
+  onEditorDidMount?: (editor: any) => void;
+
+  /** Handler to be performed when editor is blurred */
+  onBlur?: CodeEditorChangeHandler;
+
+  /** Handler to be performed when Cmd/Ctrl+S is pressed */
+  onSave?: CodeEditorChangeHandler;
+
+  /**
+   * Language agnostic suggestion completions -- typically for template variables
+   */
+  getSuggestions?: () => CodeEditorSuggestionItem[];
+}
+
+export enum CodeEditorSuggestionItemKind {
+  Method = 0,
+  Function = 1,
+  Field = 3,
+  Variable = 4,
+  Property = 9,
+  Constant = 14,
+  Keyword = 17,
+}
+
+export interface CodeEditorSuggestionItem {
```

Comment:
Very similar shape to https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/types/completion.ts#L34